VMBackup/main/safefreeze/src/safefreeze.c

```diff
@@ -119,57 +136,67 @@ int main(int argc, char *argv[])
         sigaction(SIGTTOU, &globalSignalAction, NULL)
        )
     {
-        printf("Failed to setup signal handlers\n");
+        logger("Failed to setup signal handlers\n");
         JUMPWITHSTATUS(EXIT_FAILURE);
     }
 
     for (i = 0; i < numFileSystems; i++)
     {
         char *mountPoint = argv[i + 2];
-        printf("freezing the mount: %s\n", mountPoint);
+        logger("freezing the mount: %s\n", mountPoint);
 
         if (ioctl(fileSystemDescriptors[i], FIFREEZE, 0) != 0)
         {
-            printf("Failed to FIFREEZE: %s\n", mountPoint);
+            logger("Failed to FIFREEZE: %s\n", mountPoint);
             JUMPWITHSTATUS(EXIT_FAILURE);
         }
     }
 
 
     if (kill(getppid(), SIGUSR1) != 0)
     {
-        printf("Failed to send FreezeCompletion to parent process\n");
+        logger("Failed to send FreezeCompletion to parent process\n");
         JUMPWITHSTATUS(EXIT_FAILURE);
     }
 
+    time_t starttime,currenttime; 
+    currenttime=time(NULL);
+    starttime=time(NULL);
     for (i = 0; i < timeout; i++)
     {
-        if (gThaw == 1 || sleep(1) != 0)
+        if (gThaw == 1 )
         {
             break;
         }
+        else if (sleep(1) != 0)
+        {
+           while(currenttime<starttime+i+1)
+           {
+                currenttime=time(NULL);
+           }
```

Comment:
yes I agree this is unreliable but this will be executed for the few failure case customers and not for all